### PR TITLE
Example pipeline for the readme.

### PR DESCRIPTION
**Pull request by tfenne**

### CODE REVIEW

File: pipelines/src/main/scala/dagr/pipelines/ExamplePipeline.scala [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import dagr.tasks.{DirPath, PathToFasta, PathToFastq, PathToIntervals}
+import htsjdk.samtools.SAMFileHeader.SortOrder
+
+/**
+  * Very simple example pipeline that is used in the readme.md
+  */
+@CLP(description="Example FASTQ to BAM pipeline.", group = classOf[Pipelines])
+class ExamplePipeline
+( @Arg(flag="i", doc="Input fastq.")        val fastq: PathToFastq,
+  @Arg(flag="r", doc="Reference FASTA.")    val ref: PathToFasta,
```
Comment (nh13) [inline pipelines/src/main/scala/dagr/pipelines/ExamplePipeline.scala]: `FASTA` but `fastq`.  Pick a case.

File: pipelines/src/main/scala/dagr/pipelines/ExamplePipeline.scala [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import dagr.core.cmdline.{Arg, CLP, Pipelines}
+import dagr.core.tasksystem.Pipeline
+import dagr.core.util.Io
+import dagr.tasks.bwa.BwaMem
+import dagr.tasks.picard.{CollectHsMetrics, DeleteBam, MarkDuplicates, SortSam}
+import dagr.tasks.{DirPath, PathToFasta, PathToFastq, PathToIntervals}
+import htsjdk.samtools.SAMFileHeader.SortOrder
+
+/**
+  * Very simple example pipeline that is used in the readme.md
```
Comment (nh13) [inline pipelines/src/main/scala/dagr/pipelines/ExamplePipeline.scala]: `readme` should be upper case.

File: pipelines/src/main/scala/dagr/pipelines/ExamplePipeline.scala [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  */
+@CLP(description="Example FASTQ to BAM pipeline.", group = classOf[Pipelines])
+class ExamplePipeline
+( @Arg(flag="i", doc="Input fastq.")        val fastq: PathToFastq,
+  @Arg(flag="r", doc="Reference FASTA.")    val ref: PathToFasta,
+  @Arg(flag="t", doc="Target regions.")     val targets: Option[PathToIntervals],
```
Comment (nh13) [inline pipelines/src/main/scala/dagr/pipelines/ExamplePipeline.scala]: default to `None`?

**Comment by nh13** [comment]
quibbles => :+1:

**Comment by nh13** [comment]
reminder to self: perhaps we have an `apply` method on `Pipeline` that that doesn't take an `Option[Path]` but instead `Path`?

**Comment by nh13** [comment]
for the example Pipeline, do we want to include the imports?

**Comment by nh13** [comment]
We should have a link to the `Linker` task (when ready) in the section " List of features" in the sub-section "Mechanisms for passing state"

**Comment by tfenne** [comment]
@nh13 I don't think having `Pipeline.apply(Path)` helps, because you have to chain to a constructor, not an apply method.  We could create a secondary constructor that takes `Path` instead of `Option[Path]`, but I worry that if we add more things over time (e.g. failure policy, naming context) we'll end up with dozens of constructors.

**Pull request merged**